### PR TITLE
Fix #4559 UI: Add support to get JWT token for bot users

### LINKED ISSUE #3762: Need a profile picture for ingestion bot

### LINKED ISSUE #4559: UI: Add support to get JWT token for bot users
**Feature**
Add a tab to bot user profile page for Admin users to click on a button to generate JWT tokens to be used for ingestion

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #3762
Closes #4559
### Describe your changes :

I worked on adding support for generating JWT tokens for bots users from bots details page.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :
image

image

image

image

https://user-images.githubusercontent.com/59080942/167467390-228c5270-d273-400a-a1cb-f46f1902e5e6.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

**Comment by harshach** [comment]
@Sachin-chaurasiya can we get away from dialog boxes . We need to stop using dialog boxes as the styling element for inputs
image
cc @vivekratnavel @devyani-kaushik

**Comment by vivekratnavel** [comment]
I agree with moving away from dialog boxes. Also, I think the JWT token field doesn't have to be a password field.

File: openmetadata-ui/src/main/resources/ui/src/components/BotsDetail/BotsDetail.component.tsx [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+  const fetchRightPanel = () => {
+    return (
+      <div data-testid="right-panel">
+        <div className="tw-pb-4 tw-mb-4 tw-border-b tw-flex tw-flex-col">
+          <h6 className="tw-mb-2 tw-text-lg">Token Security</h6>
+          <p className="tw-mb-2">
+            Anyone who has your REST API key will be able to send notifications
+            from your app. Do not expose the REST API key in your application
+            code. Do not share it on GitHub or anywhere else online.
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/BotsDetail/BotsDetail.component.tsx:214]:
```suggestion
            Anyone who has your JWT Token will be able to send REST API requests to the OpenMetadata Server. Do not expose the JWT Token in your application
            code. Do not share it on GitHub or anywhere else online.
```

File: openmetadata-ui/src/main/resources/ui/src/components/BotsDetail/BotsDetail.component.tsx [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+          onCancel={() => setIsRevokingToken(false)}
+          onConfirm={() => {
+            revokeTokenHandler();
+            setIsRevokingToken(false);
+          }}
+        />
+      ) : null}
+      {isRegeneratingToken ? (
+        <ConfirmationModal
+          bodyText="The re-generating token will rewrite the existing JWT token"
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/BotsDetail/BotsDetail.component.tsx:312]:
```suggestion
          bodyText="Generating a new token will revoke the existing JWT token. Are you sure you want to proceed?"
```

File: openmetadata-ui/src/main/resources/ui/src/components/BotsList/BotsList.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            </span>
+          )}
+        </div>
+      </div>
+    );
+  };
+
+  const getListComponent = () => {
+    if (!bots.length) {
+      return <ErrorPlaceHolder>No bots user available</ErrorPlaceHolder>;
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/BotsList/BotsList.tsx:66]:
```suggestion
      return <ErrorPlaceHolder>No bots are available</ErrorPlaceHolder>;
```

File: openmetadata-ui/src/main/resources/ui/src/components/Modals/GenerateTokenModal/GenerateTokenModal.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+  return (
+    <dialog
+      className={classNames('tw-modal', className)}
+      data-testid="generate-token-modal">
+      <div className="tw-modal-backdrop" />
+      <div className="tw-modal-container tw-w-120">
+        <div className={classNames('tw-modal-header')}>
+          <p className="tw-modal-title" data-testid="modal-header">
+            New JWT token
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/components/Modals/GenerateTokenModal/GenerateTokenModal.tsx:87]:
```suggestion
            Generate JWT token
```

**Review by vivekratnavel — APPROVED** [review]

**Pull request merged**